src/main/java/com/limechain/rpc/methods/state/StateRPCImpl.java

```diff
@@ -264,19 +263,19 @@ public Map<String, Object> stateGetReadProof(final List<String> keyHexList, fina
         final byte[] blockStateRoot = blockState.getBlockStateRoot(blockHash).getBytes();
 
         List<String> readProof = keyHexList
-            .stream()
-            .map(StringUtils::hexToBytes)
-            // TODO change implementation logic as this is not correct
-            .map(key -> trieStorage.getByKeyFromMerkle(blockStateRoot, Nibbles.fromBytes(key))
-                .orElse(null))
-            .filter(Objects::nonNull)
-            .map(NodeData::getMerkleValue)
-            .map(StringUtils::toHexWithPrefix)
-            .toList();
+                .stream()
+                .map(StringUtils::hexToBytes)
+                // TODO change implementation logic as this is not correct
```

Comment:
I know you didn't leave the comment but can you update it to point at the incorrect part